lib/utilities/key_exists.py

```diff
@@ -0,0 +1,17 @@
+
+def key_exists(search_key, inputed_dict):
```

Comment:
Now instead of having multiple if statements when searching for 'details.<somekey>', we can use this utility function to make things cleaner.